### ISSUE TITLE
TitleBar control forces fixed height in its UpdateSize method

### ISSUE THREAD

**Issue by billhenn** [AvaloniaUI/Avalonia#11335]
We are doing a custom theme for the `TitleBar` control and noticed that after the window state changes to `Maximized`, the title bar forces a short fixed height.  We are trying to make a title bar that has a larger-than-normal height due to its other content, similar to Office and Visual Studio.  What ends up happening is the content is all clipped when the window is maximized.

This screenshot shows the difference in height.  The top half is the window when in a default restored state and the bottom half shows when maximized.

![image](https://github.com/AvaloniaUI/Avalonia/assets/5760058/4854e353-6eb4-4044-9d0b-056fce15eb99)

The bottom seems to force a system-defined title bar height instead of simply saying use the system-defined title bar height as a "minimum" height.

This is where the problematic code begins:
https://github.com/AvaloniaUI/Avalonia/blob/31e2f3375c8b913f9d02b3f858994540eef6ccc6/src/Avalonia.Controls/Chrome/TitleBar.cs#L28

It should set `MinHeight` instead of `Height`, and also should NOT be setting the `CaptionButtons.Height` at all.  I want those to fully match the height of the title bar, like in Office/VS.  If I wanted to align them top, I could do so in the templates instead.

To summarize, it's fine to set the `TitleBar.MinHeight` (a coerce would be even better if the a developer like us wanted their own `MinHeight`) but you should not set a fixed height.  And caption buttons should not be updated at all since that can be done in the template where we have control over things.

**Comment by timunie** [comment]
Good catch. I think a PR for this would be welcome. Are you willing to provide a PR? coercing should work well here taking MinHeight into account.
